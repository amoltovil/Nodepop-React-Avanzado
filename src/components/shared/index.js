export { default as Button } from './Button';
export { default as FormField } from './FormField';
export { default as Photo } from './Photo';
export { default as Textarea } from './Textarea';
export { default as NotFoundPage } from './NotFoundPage';
export { default as Confirmation } from './Confirmation';

//export { default as TagsList } from './TagsList';