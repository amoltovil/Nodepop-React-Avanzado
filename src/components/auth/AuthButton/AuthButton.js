import React from 'react';
import T from 'prop-types';
import { Link } from 'react-router-dom';
import { logout } from '../../../api/auth';
import { Confirmation } from '../../shared';
import { connect } from 'react-redux';
import { getIsLogged } from '../../../store/selectors';
import { authLogout } from '../../../store/actions';

const AuthButton = ({ className, isLogged, onLogout}) => {
  
  // const handleLogoutClick = () => {
  //   // Confirmación de logout  (anterior)
  //   const res = window.confirm("Confirme que desea salir de la aplicación Nodepop");
  //   if (res == true) {
  //     logout().then(onLogout);
  //   }
  // };

  const handleLogoutConfirm = async () => {
    await logout().then(onLogout);    
  };

  return isLogged ? (
    <Confirmation
      message="Confirme que desea salir de la aplicación Nodepop"
      onConfirm={handleLogoutConfirm}
    >
      Logout
    </Confirmation>
  ) : (
    <Link to="/login">Login</Link>
  );
};

AuthButton.propTypes = {
  className: T.string,
  isLogged: T.bool,
  onLogout: T.func.isRequired,
};

AuthButton.defaultProps = {
  isLogged: false,
};

// const ConnectedAuthButton = props => {
//   return (
//     <AuthContextConsumer>
//       {value => {
//         return (
//           <AuthButton
//             isLogged={value.isLogged}
//             onLogout={value.onLogout}
//             {...props}
//           />
//         );
//       }}
//     </AuthContextConsumer>
//   );
// };

// export default ConnectedAuthButton;

//const mapStateToProps = (state, ownProps) => console.log(ownProps) || { isLogged: getIsLogged(state) };
const mapStateToProps = (state, ownProps) => ({ isLogged: getIsLogged(state) });

const mapDispatchToProps = dispatch => ({
  onLogout: () => dispatch(authLogout()), 
});

// es lo mismo que lo de arriba, es un objecto donde si las acciones 
// const mapDispatchToProps = {
//   onLogout: authLogout,
// };

// const mapDispatchToProps = {
//   onLogout,
//}

//const mapDispatchToProps = authActions;

export default connect(mapStateToProps, mapDispatchToProps)(AuthButton);