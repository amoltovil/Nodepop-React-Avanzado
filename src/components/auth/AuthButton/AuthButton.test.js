import React from 'react';
import { shallow } from 'enzyme';
import { Confirmation } from '../../shared';
//import { AUTH_LOGOUT, } from '../../../store/types';
//import { authLogout } from '../../../store/actions';

import AuthButton from './AuthButton';

describe('AuthButton', () => {
    const props = {
        className: '',
        isLogged: true,
        onLogout: jest.fn(),
        onConfirm: jest.fn(), 
    };
    
    const api = {
        auth: { logout: jest.fn().mockResolvedValue() },
      };

    const dispatch = jest.fn()

    const render = () => shallow(<Confirmation {...props} />);

    test('should render', () => {
        const wrapper = render();
        //console.log(wrapper.props());
        expect(wrapper.exists()).toBe(true);
        //expect(wrapper.find('onConfirm').props()).toBe('onConfirm');
        //expect(wrapper.props().isLogged).to.equal(true);
        //wrapper.find('Confirmation').invoke('onConfirm')().then(() => {
          //  expect().toBeCalled();
        //});
    });

    test('snapshot testing', () => {
        const wrapper = render();
        expect(wrapper).toMatchSnapshot();
    });

});
