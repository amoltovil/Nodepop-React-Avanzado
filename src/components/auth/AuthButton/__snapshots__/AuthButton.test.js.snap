// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButton snapshot testing 1`] = `
<Fragment>
  <styled.button
    className=""
    confirmation={null}
    isLogged={true}
    onClick={[Function]}
    onLogout={[MockFunction]}
  />
</Fragment>
`;
